libraries/core_libs/consensus/src/transaction_manager/transaction_manager.cpp

```diff
@@ -254,26 +233,8 @@ size_t TransactionManager::getNonfinalizedTrxSize() const {
  * Retrieve transactions to be included in proposed block
  */
 SharedTransactions TransactionManager::packTrxs(uint16_t max_trx_to_pack) {
-  SharedTransactions to_be_packed_trx;
-  if (max_trx_to_pack == 0) {
-    to_be_packed_trx = getTransactionsSnapShot();
-  } else {
-    to_be_packed_trx.reserve(max_trx_to_pack);
-    {
-      uint16_t count = 0;
-      std::shared_lock transactions_lock(transactions_mutex_);
-      for ([[maybe_unused]] const auto &[hash, trx] : transactions_pool_) {
-        if (max_trx_to_pack == count) break;
-        to_be_packed_trx.push_back(trx);
-        count++;
-      }
-    }
-  }
-
-  // sort trx based on sender and nonce
-  std::sort(to_be_packed_trx.begin(), to_be_packed_trx.end(), trxComp);
-
-  return to_be_packed_trx;
+  std::shared_lock transactions_lock(transactions_mutex_);
```

Comment:
I think all locking that was protecting previous tx_pool should be moved into the TransactionQueue class

Reply:
this is not possible because transactions_mutex_ is locking more then to this pool

Reply:
it is used for DB and pool synchronization and many more

Reply:
in such case it would be nice to have a comment in tx_pool header explaining why it does not use locks